Update instructions for adding an expiring soon domain (#3869)
cisagov/getgov: docs/developer/registry-access.md

This is helpful for debugging situations where epplib is not correctly or fully parsing the XML returned from the registry.

### Adding in an 'expiring soon' or 'expired' domain

> Expiration dates created by fixtures in sandboxes are not accurate (it resets to the expiration date to a year from 'today').

Author: @lizpearl - are you able to add more context to explain why this is?

> In order to view an expiring or expired domain, we need to pull a record from staging and use it on the desired sandbox. In order to have edit ability, you must be a domain manager of that domain. Below are instructions on how to add a domain from staging sandbox into another sandbox and how to be able to view the domain. As a domain manager you will be able to view the renewal form.

Author: @lizpearl - can you explain how this works? I would expect the db to be totally separate between sandboxes. Why does creating (or getting?)  a domain to my sandbox with the same name as a domain in staging access the record?

> 

Reviewer: Possible thought: Adding a subheading here to indicate what certain steps are doing (eg: Find an expiring domain in staging or something along the lines of that)

Author: Good idea- added one

1. Go to the `staging` sandbox and to `/admin`
2. Go to Domains and find a domain that is actually expired or expiring soon by sorting the Expiration Date column

> 3. Click into the domain to check the expiration date

Reviewer: Curious why we need to check the expiration date specifically here. Is there an expectation that the domain expiration date on the domain record may not match the expiration date on the table?

4. Click into Manage Domain to double check the expiration date as well (this expiration date is the source of truth)
5. Now hold onto that domain name, and save it for the command below

> 

Reviewer: Possible thought: Adding a subheading here to indicate the intention of these below steps (eg: Duplicate(?) the domain into your sandbox environment of choice)

Author: Done

> 6. In a terminal, run these commands:

Reviewer: Possible thought: Adding a subheading here to indicate the intention of these below steps (eg: Duplicate(?) the domain into your sandbox environment of choice)

> 6. In a terminal, run these commands:

Reviewer: I think we should delineate from here you’re going to be copying over this production domain into your sandbox. The terminal commands do mention your initials are there but it’s unclear otherwise

```

> cf ssh getgov-<your-intials>

Reviewer: ```suggestion
cf ssh getgov-<environment-name>
```

6. In a terminal, run these commands:
```
cf ssh getgov-<your-intials>
/tmp/lifecycle/shell
./manage.py shell
from registrar.models import Domain, DomainInvitation
from registrar.models import User
user = User.objects.filter(first_name="<your-first-name>")
from registrar.models import Domain
domain = Domain.objects.get_or_create(name="<that-domain-here>")
```

> 7. Go back to `/admin` and create Domain Information for that domain you just added in via the terminal 

Reviewer: Clarifying question: What is the specific reason we're creating a Domain information object for this domain? I know Domain object needs Domain Information associated with it, but would like specifics — is the Manage Domain page dependent on the Domain having a Domain Information object?
Also, what fields do you need to fill out for this Domain Information object. Doesn’t make sense to assign it a Domain Request if we never made one from directly creating this Domain object. Feel free to verify if any of these fields are necessary (ideally we want to fill out the bare minimum Domain Information fields to create a Domain Info object):
    - Portfolio if org model
    - Creator?
    - Domain

> 8. Go to Domain to find it 

Reviewer: clarifying question: what are we finding - the domain we just created? If so, we could reword this to "Locate the Domain record page of the domain you just created" or something along the lines of that

> 9. Click Manage Domain

Reviewer: Clarifying question: Why do you need to add yourself as a domain manager from the Manage Domain page as opposed to adding yourself by adding a User Domain Role object from /admin?

Reviewer 2: > Why do you need to add yourself as a domain manager from the Manage Domain page as opposed to adding yourself by adding a User Domain Role object from /admin?

Great call out, you can do so both ways

> 11. Go to the Registrar page and you should now see the expiring domain

Reviewer: ```suggestion
11. You should now be able to access your expiring domain in the registrar.
```

> To view the domain in the org model, it must be associated with a portfolio and viewed from within that portfolio. If the domain is not already part of a portfolio, in Django Admin, go to Domains, find the domain, and update the "Requested by - Portfolio" field to a portfolio of your choice.

Reviewer: As org model becomes more frequently used, what are people's thoughts on if we should just include how to set up an expiring/expired domain in org model in the order where it's relevant? ie somewhere around step ~7/8, mentioning if the domain is in org model that they should add do the above steps in admin
